tests/requests/tests.py

```diff
@@ -758,7 +758,7 @@ def test_host_validation_in_debug_mode(self):
         If ALLOWED_HOSTS is empty and DEBUG is True, variants of localhost are
         allowed.
         """
-        valid_hosts = ['localhost', '127.0.0.1', '[::1]']
+        valid_hosts = ['foo.localhost', 'bar.localhost', 'baz.localhost', 'localhost', '127.0.0.1', '[::1]']
```

Comment:
These entries are redundant; they cover the same line of code. A single one should suffice

```suggestion
        valid_hosts = ['localhost', 'subdomain.localhost', '127.0.0.1', '[::1]']
```